### ISSUE TITLE
[Issue] Load Html Sometimes Doesnt Load a body

### ISSUE THREAD

**Issue by BrandonMayjeye** [Voltstro-Studios/UnityWebBrowser#351]
### What platform are you experiencing this issue on?

Linus (Arch)
### What version of UWB are you using?

2.2.1

### What Unity version are you running?

2022.3.48f1

### Describe what the issue you are experiencing is.

The html load function does not properly load some html data
it works sometimes but this specific set doesnt work well.

### Provide reproducible steps for this issue.

Call browser LoadHtml(string) and put in the below html data.

### Any additional info you like to provide?

```html

<html>
<style>

logentry {
    /* display: block; */
    border: 1px solid black;
    font-size: 2em;
    display: grid;
    box-shadow: 4px 5px 6px black;
    margin-bottom: 1em;
    background: #c5c3c3;
}

guid {
    display: none;
}

breadcrumb {
    display: inline-flex;
    border-bottom: 1px solid black;
    cursor: pointer;
}

playername {
    display: none;
}

aliasname {
    color: red;
    font-weight: bolder;
}

log {
    font-weight: 100;
    font-style: italic;
}



logentry:hover {
    /* background: #959494; */
}

breadcrumb:hover {
    background: #898989;
}

</style>
<body>
<h1>Log Entries for DNS</h1>
<logEntries>
<logEntry><breadcrumb onclick='uwb.ExecuteJsMethod(`navigate`, `[Redacted]?searchTerm=5d9b73e6-01d7-46e5-a4e6-6cacac98843d`);' breadcrumb.guid='5d9b73e6-01d7-46e5-a4e6-6cacac98843d'>
            <guid>5d9b73e6-01d7-46e5-a4e6-6cacac98843d</guid>
            <time>10/15/2024 1:09:15 PM</time>
            <name>Dns:IsAddressValid</name>
            <playerName>[Redacted]</playerName>
            <aliasName>[Redacted]</aliasName></breadcrumb><log> Requests to see if [Redacted]is valid in address book valueOf Operation is True</log></logEntry>
<logEntry><breadcrumb onclick='uwb.ExecuteJsMethod(`navigate`, `[Redacted]?searchTerm=d0c28bce-b0cf-44ad-820c-34b8b21a50ea`);' breadcrumb.guid='d0c28bce-b0cf-44ad-820c-34b8b21a50ea'>
            <guid>d0c28bce-b0cf-44ad-820c-34b8b21a50ea</guid>
            <time>10/15/2024 1:09:24 PM</time>
            <name>Dns:IsAddressValid</name>
            <playerName>[Redacted]</playerName>
            <aliasName>[Redacted]</aliasName></breadcrumb><log> Requests to see if [Redacted] is valid in address book valueOf Operation is True</log></logEntry>

</logEntries>
</body>
</html>
```

* Html output using dev tools
```html
<html><head><style>

logentry {
    /* display: block; */
    border: 1px solid black;
    font-size: 2em;
    display: grid;
    box-shadow: 4px 5px 6px black;
    margin-bottom: 1em;
    background: </style></head><body></body></html>
```

**Comment by BrandonMayjeye** [comment]
Further information 
removing the styles from the html loads the body,m
adding styles back in remove the body
heres my text implementation
$styleCss injects the css into the text
and $entries injects the log entries that dont appear when css is on


```html
<html>
<head>
<style>
$styleCss
</style>
</head>
<body>
<h1>Log Entries for DNS</h1>
<logEntries>
$entries
</logEntries>
</body>
</html>
```

**Comment by BrandonMayjeye** [comment]
another experiment moving the css style to the endof of the body
loads the $entries and part of the css

maybe theres a limit to the bytes that can be transmitted to the browser



```html
<html><head></head><body>


<h1>Log Entries for DNS</h1>
<logentries>
<logentry><breadcrumb onclick="uwb.ExecuteJsMethod(`navigate`, `[redacted].com?searchTerm=0f0b3410-3c2b-4173-97aa-1695d6e06b2d`);" breadcrumb.guid="0f0b3410-3c2b-4173-97aa-1695d6e06b2d">
            <guid>0f0b3410-3c2b-4173-97aa-1695d6e06b2d</guid>
            <time>10/15/2024 1:31:30 PM</time>
            <name>Dns:IsAddressValid</name>
            <playername>[redacted]</playername>
            <aliasname>[redacted]</aliasname></breadcrumb><log> Requests to see if [redacted].dev is valid in address book valueOf Operation is True</log></logentry>
<logentry><breadcrumb onclick="uwb.ExecuteJsMethod(`navigate`, `[redacted]?searchTerm=70142b92-5fb1-403f-b8b5-3fe32689e4ff`);" breadcrumb.guid="70142b92-5fb1-403f-b8b5-3fe32689e4ff">
            <guid>70142b92-5fb1-403f-b8b5-3fe32689e4ff</guid>
            <time>10/15/2024 1:31:41 PM</time>
            <name>Dns:IsAddressValid</name>
            <playername>[redacted]</playername>
            <aliasname>[redacted]</aliasname></breadcrumb><log> Requests to see if dns.com is valid in address book valueOf Operation is True</log></logentry>

</logentries>



<style>

logentry {
    /* display: block; */
    border: 1px solid black;
    font-size: 2em;
    display: grid;
    box-shadow: 4px 5px 6px black;
    margin-bottom: 1em;
    background: </style></body></html>
    
    ```



```html
        private string logHtml = @"
<html>

<body>


<h1>Log Entries for DNS</h1>
<logEntries>
$entries
</logEntries>



<style>
$styleCss
</style>


</body>
</html>
";

```

**Comment by Voltstro** [comment]
`LoadHtml` currently just calls `LoadUrl` with a url of `data:text/html,{html}`.

The hashtags in your css for your colours are causing the url to be fragmented by the browser, so the browser only renders the content upto the hashtag.

A workaround for now is to just use rgb() in your css.

This ideally shouldn't be a problem. CEF doesn't provide a direct way to load html. I will have to look into a better solution for loading html.

**Comment by Voltstro** [comment]
Having a look at the cef examples, it seems like they base64 encode then url encode the provided html content before calling load url with `data:text/html;base64,{html}`.

I have done this and it fixes the issues of the hashtags fragmenting the url which causes the rendering issues. Fix will come in next release.

**Comment by Voltstro** [comment]
Fixed in release 2.2.2.

**Comment by BrandonMayjeye** [comment]
Thanks man, its working great now , I really appreciate it
